testing/upload_validation.ts

```diff
@@ -999,9 +1000,11 @@ ${endpointKey ? 'endpointKey is set' : `requiresEndpointUrl is ${requiresEndpoin
     isSystem: z.boolean().optional(),
     extraOAuthScopes: z.array(z.string()).optional(),
     allowedAuthenticationNames: z.array(z.string()).optional(),
+    // Has to be any to avoid circular dependency.
+    validateParameters: z.any().optional(),
```

Comment:
Hmm we should play with this a bit after you're unblocked. I think we should separate out a base schema that doesn't include `validateParameters`, so we can properly validate this. With this change we're also actually allowing you to recursively put validateParameters on validateParameters, which I don't think makes sense either.